content/docs/use-cases/ci-cd-for-machine-learning.md

```diff
@@ -0,0 +1,98 @@
+# Continuous Integration and Deployment for Machine Learning
+
+Applying DevOps methodologies to machine learning (MLOps) and data management
+(DataOps) is increasingly common. This means resource orchestration
+(provisioning servers for model training), model testing (validating model
+inference), and model deployment to production, as well as monitoring &
+feedback. [DVC](/) can manage data/models and reproducible pipelines, while
+[CML] can assist with orchestration, testing and monitoring.
+
+[cml]: https://cml.dev
+
+![](https://static.iterative.ai/img/ml-pipeline.png) _Basic ML pipeline_
+
+The main benefits of CI/CD (continuous integration and continuous delivery) for
+ML are detailed below.
+
+Firstly, we can automate and enforce testing. This includes enforcing integrity
+with application-specific tests:
+
+- **Data validation**: for example, validation against a schema or verifying
+  pipeline consistency -- correct shapes, data types, etc.
+- **Model validation**: for example, input/output and performance validation --
+  all dependencies present for inference to run, and model scores within
+  thresholds.
+
+Typically, this requires **infrastructure orchestration**: making CI systems
+provision and launch a GPU instance, train a model, cleanly terminate, and pull
+results back. In particular:
+
+- **Provisioning**: When data or code is added/changed in a pull request (PR),
+  automatically running validation tests in the cloud.
+- **Metrics**: Automatically generating metrics reports.
+
+Finally, this can be taken a step further with entire **experiments**/research
+being run online:
+
+- Fine-tune on a schedule: set up jobs to pull in new data from regularly
+  updated external sources to retrain/refine deployed models.
+- Run a hyperparameter search: keep heavyweight resources and big data on CI
+  servers running overnight, and put your laptop to sleep.
+
+Normally, CI/CD is hard to set up, configure, and maintain -- especially for
+data and ML pipelines.
+
+![](https://static.iterative.ai/img/ml-vs-cicd.png) _Traditional ML meets CI/CD_
```

Comment:
I think this can be the first image since from the get go we're talking about CI/CD. The current first figure doesn't connect clearly to the intro.

![image](https://user-images.githubusercontent.com/1477535/128110494-a3a56b2b-ee0a-4cb8-8ce8-5bb1d33ea726.png)

For a 2nd figure (or more) the scheme could be simplified/enhanced with DVC and CML.